aws-distro-opentelemetry-node-autoinstrumentation/test/patches/instrumentation-patch.test.ts

```diff
@@ -692,6 +693,60 @@ describe('InstrumentationPatchTest', () => {
       });
     });
 
+    it('Prevents recursion when credentials provider makes STS calls', async () => {
+      // Track credentials provider calls and skip flag state
+      let credentialsProviderCallCount = 0;
+      let skipCredentialCaptureValue = false;
+
+      // Mock span for attribute capture
+      const mockSpan = { setAttribute: sinon.stub() };
+      sinon.stub(trace, 'getSpan').returns(mockSpan as unknown as Span);
+
+      // Patch ROOT_CONTEXT to capture setValue calls
+      const originalSetValue = ROOT_CONTEXT.setValue;
+      ROOT_CONTEXT.setValue = (key: symbol, value: any) => {
+        if (key.toString().includes('skip-credential-capture')) {
+          skipCredentialCaptureValue = value;
+        }
+        return ROOT_CONTEXT;
+      };
+      sinon.stub(context, 'active').returns(ROOT_CONTEXT);
+
+      // Capture middleware added to stack
+      const middlewareStack: any[] = [];
+      const recursiveSdkSend = extractAwsSdkInstrumentation(PATCHED_INSTRUMENTATIONS)
+        ['_getV3SmithyClientSendPatch'](() => Promise.resolve())
+        .bind({
+          middlewareStack: { add: (middleware: any, config: any) => middlewareStack.push([middleware, config]) },
+          config: {
+            // Credentials provider that recursively calls SDK (simulates STS)
+            credentials: async () => {
+              credentialsProviderCallCount++;
+              await recursiveSdkSend({}, null);
+              return { accessKeyId: 'test-access-key' };
+            },
+            region: () => Promise.resolve('us-west-2'),
+          },
+        });
+
+      // Initial SDK call triggers middleware setup
+      await recursiveSdkSend({}, null);
+
+      // Execute credentials extraction middleware
+      await middlewareStack[1][0](() => Promise.resolve(), null)({});
```

Comment:
As discussed, is it feasible to perform this `Execute credentials extraction middleware` through a realistic AWS SDK Client? I think you'd need to somehow patch the `config.credentials` method of the AWS SDK Client to perform another AWS SDK call.

This will also test that the mitigation applies when in a new Context, in which the middleware's AWS SDK call should be in.